spec/models/journable/with_historic_attributes_spec.rb

```diff
@@ -0,0 +1,517 @@
+        end
+      end
+    end
+
+    context "with an AR of work packages" do
+      let(:work_packages) { WorkPackage.all }
+
+      describe "when providing a query" do
+        it "determines for each timestamp whether the journables matches the query at that timestamp" do
+          expect(subject.first.matches_query_filters_at_timestamps).to include Timestamp.parse("2022-01-01T00:00:00Z")
+          expect(subject.first.matches_query_filters_at_timestamps).not_to include Timestamp.parse("PT0S")
+          expect(subject.last.matches_query_filters_at_timestamps).to include Timestamp.parse("2022-01-01T00:00:00Z")
+          expect(subject.last.matches_query_filters_at_timestamps).not_to include Timestamp.parse("PT0S")
+        end
+      end
+    end
+  end
+
+  describe "#baseline_timestamp" do
+    let(:journable) { described_class.wrap(work_package1, timestamps:) }
+
+    subject { journable.baseline_timestamp }
+
+    it "provides simplified access to the baseline timestamp, which is the first given timestamp" do
+      expect(subject).to eq Timestamp.parse("2022-01-01T00:00:00Z")
+    end
+  end
+
+  describe "#baseline_attributes" do
+    let(:journable) { described_class.wrap(work_package1, timestamps:) }
+
+    subject { journable.baseline_attributes }
+
+    it "provides access to the work-package attributes at the baseline timestamp" do
+      expect(subject.subject).to eq "The original work package 1"
+    end
+
+    describe "when the work package did not exist yet at the baseline date" do
+      let(:timestamps) { [Timestamp.parse("2021-01-01T00:00:00Z"), Timestamp.parse("PT0S")] }
+
+      it "has no baseline attributes" do
+        expect(subject).to be_nil
+      end
+    end
+  end
+
+  describe "#matches_query_filters_at_baseline_timestamp?" do
+    let(:journable) { described_class.wrap(work_package1, timestamps:, query:) }
+    let(:query) { build_query }
+
+    subject { journable.matches_query_filters_at_baseline_timestamp? }
+
+    describe "providing a filter that matches at the baseline timestamp" do
+      let(:search_term) { "original" }
+
+      it "determines whether the journable matches the query at the baseline timestamp" do
+        expect(subject).to be true
+      end
+    end
+
+    describe "providing a filter that matches at the current timestamp" do
+      let(:search_term) { "current" }
+
+      it "determines whether the journable matches the query at the baseline timestamp" do
+        expect(subject).to be false
+      end
+    end
+  end
+
+  describe "#current_timestamp" do
+    let(:journable) { described_class.wrap(work_package1, timestamps:) }
+
+    subject { journable.current_timestamp }
+
+    it "provides simplified access to the current timestamp, which is the last given timestamp" do
+      expect(subject).to eq Timestamp.parse("PT0S")
+    end
+  end
+
+  describe '.changed_at_timestamp' do
```

Comment:
This should be `#changed_at_timestamp` because this is an instance method.

Reply:
Fixed in 5431eb0780